thirdparty/README.md

```diff
@@ -827,6 +827,13 @@ and solve conflicts and also enrich the feature set originally
 proposed by these libraries and better integrate them with Godot.
 
 
+## spirv-cross
+
+- Upstream: https://github.com/KhronosGroup/SPIRV-Cross
+- Version: main (b8fcf307f1f347089e3c46eb4451d27f32ebc8d3, 2024)
```

Comment:
I think it make sense to track the tag which corresponds to the Vulkan SDK we're using, like we do for other Vulkan/SPIR-V components.

See the diff below, also adding instructions on how to update the library.


Based on these instructions, you can also remove the following unused files:
- `README.md`
- `LICENSES/CC-BY-4.0.txt` (only applies to the README)
- `spirv.h` (only used by `spirv_cross_c.h`)

```diff
diff --git a/thirdparty/README.md b/thirdparty/README.md
index 5418ac8e77..35cadf973a 100644
--- a/thirdparty/README.md
+++ b/thirdparty/README.md
@@ -830,9 +830,18 @@ proposed by these libraries and better integrate them with Godot.
 ## spirv-cross
 
 - Upstream: https://github.com/KhronosGroup/SPIRV-Cross
-- Version: main (b8fcf307f1f347089e3c46eb4451d27f32ebc8d3, 2024)
+- Version: vulkan-sdk-1.3.283.0 (2ccc81fd826e4dd4a2db2f94b8e6eb738a89f5f1, 2024)
 - License: Apache 2.0
 
+Version should be kept in sync with the one of the used Vulkan SDK (see `vulkan`
+section).
+
+Files extracted from upstream source:
+
+- All `.cpp`, `.hpp` and `.h` files, minus `main.cpp`, `spirv_cross_c.*`, `spirv_hlsl.*`
+- `include/` folder
+- `LICENSE` and `LICENSES/` folder, minus `CC-BY-4.0.txt`
+
 
 ## spirv-reflect
 
@@ -954,6 +963,7 @@ to tagged SDK releases. All Vulkan libraries and headers should be kept in sync
 - Update Vulkan SDK components to the matching tag (see "vulkan")
 - Update volk (see "volk")
 - Update glslang (see "glslang")
+- Update spirv-cross (see "spirv-cross")
 - Update spirv-reflect (see "spirv-reflect")
 
 Files extracted from upstream source:
```

Reply:
Grabbed everything except adding it to the `vulkan` section, as it doesn't have to be tied to the `vulkan` version. `spirv-cross` is used to generate MSL, and therefore we might need to use newer versions to fix MSL code generation issues, which happens to be the case to resolve some problems for iOS.